### ISSUE TITLE
eval_mmlu ?

### ISSUE THREAD

**Issue by mistletoe1024** [mobiusml/hqq#136]

**Comment by mistletoe1024** [comment]
bug fixed.

**Comment by mobicham** [comment]
Hi, sorry for the delay. What was the problem again ?

**Comment by mistletoe1024** [comment]
Hi, I'd like to evaluate the performance of hqq+ on mmlu, 
so I writed a eval_mmlu() behind eval_wikitext2() in hqq_plus.py 
referred to  https://github.com/FranxYao/chain-of-thought-hub/blob/main/MMLU/run_mmlu_llama.py, 
but I got a bad avg score (only 0.0438:(, but FP16 is  0.4583),
my quantization setting is "nbits=2, group_size=8, quant_scale=False, quant_zero=False, axis=0",
lora_params and training_args were not change. 
Could you offer a better solution or any advice? thank you!

-----------------------------------------------------
Today, I printed the pred_answer of  eval_mmlu() after eval_wikitext2() ,  and found lots of '\n' in the list of pred_answer, I thought i had solved the problem, but still got bad score of mmlu (0.0342, and ppl is  5.601, seems normal). So could you offer your solution?

**Comment by mobicham** [comment]
Oh I see, which model exactly? Is this an issue with MMLU only or also the other metrics?

**Comment by mistletoe1024** [comment]
> Oh I see, which model exactly? Is this an issue with MMLU only or also the other metrics?

Llama-2-7b-hf， only MMLU at present, the ppl is normal.

**Comment by mobicham** [comment]
Did you train the model yourself? How did you train it (what dataset) etc.

**Comment by mistletoe1024** [comment]
> Did you train the model yourself? How did you train it (what dataset) etc.

yes, but i just ran the hqq_puls.py and did not change any params. So the dataset is dataset = load_dataset('wikitext', 'wikitext-2-raw-v1', split='train')

**Comment by mobicham** [comment]
Sounds like it's overfiting to wikitext, that might explain why the ppl is good but mmlu is bad. You need proper instruct data (and the instruct model not base model) to get good mmlu performance, which is outside the scope of the hqq+ script

**Comment by mistletoe1024** [comment]
> Sounds like it's overfiting to wikitext, that might explain why the ppl is good but mmlu is bad. You need proper instruct data (and the instruct model not base model) to get good mmlu performance, which is outside the scope of the hqq+ script

Ok, I am going to try other dataset or instruct model, thank you :)

**Comment by mobicham** [comment]
It's normally a mix of datasets not just 1. It's an art to figure out the right datasets :D 
Let me know how that works out!
